Can't install on ubunto16.04

when I execute ./xtuple-utility.sh -a, occurs error:
make: Leaving directory '/opt/xtuple/4.10.1/xtuple-web/xtuple/node_modules/socket.io/node_modules/socket.io-client/node_modules/ws/build'
npm WARN engine escodegen@1.9.0: wanted: {"node":">=0.12.0"} (current: {"node":"0.10.40","npm":"1.4.28"})
npm WARN engine request@2.83.0: wanted: {"node":">= 4"} (current: {"node":"0.10.40","npm":"1.4.28"})
npm WARN engine request-promise-native@1.0.5: wanted: {"node":">=0.12.0"} (current: {"node":"0.10.40","npm":"1.4.28"})
npm WARN engine esprima@3.1.3: wanted: {"node":">=4"} (current: {"node":"0.10.40","npm":"1.4.28"})
npm WARN engine har-validator@5.0.3: wanted: {"node":">=4"} (current: {"node":"0.10.40","npm":"1.4.28"})
npm WARN engine form-data@2.3.1: wanted: {"node":">= 0.12"} (current: {"node":"0.10.40","npm":"1.4.28"})
npm WARN engine hawk@6.0.2: wanted: {"node":">=4.5.0"} (current: {"node":"0.10.40","npm":"1.4.28"})
npm WARN engine punycode@2.1.0: wanted: {"node":">=6"} (current: {"node":"0.10.40","npm":"1.4.28"})
npm WARN engine har-schema@2.0.0: wanted: {"node":">=4"} (current: {"node":"0.10.40","npm":"1.4.28"})
npm WARN engine cryptiles@3.1.2: wanted: {"node":">=4.0.0"} (current: {"node":"0.10.40","npm":"1.4.28"})
npm WARN engine sntp@2.1.0: wanted: {"node":">=4.0.0"} (current: {"node":"0.10.40","npm":"1.4.28"})
npm WARN engine hoek@4.2.0: wanted: {"node":">=4.0.0"} (current: {"node":"0.10.40","npm":"1.4.28"})
npm WARN engine boom@4.3.1: wanted: {"node":">=4.0.0"} (current: {"node":"0.10.40","npm":"1.4.28"})
npm WARN engine boom@5.2.0: wanted: {"node":">=4.0.0"} (current: {"node":"0.10.40","npm":"1.4.28"})
npm WARN engine co@4.6.0: wanted: {"iojs":">= 1.0.0","node":">= 0.12.0"} (current: {"node":"0.10.40","npm":"1.4.28"})
npm ERR! 404 Not Found
npm ERR! 404 
npm ERR! 404 'types/node' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it
npm ERR! 404 It was specified as a dependency of 'parse5'
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, or http url, or git url.

npm ERR! System Linux 4.10.0-28-generic
npm ERR! command "node" "/usr/local/bin/npm" "install"
npm ERR! cwd /opt/xtuple/4.10.1/xtuple-web/xtuple
npm ERR! node -v v0.10.40
npm ERR! npm -v 1.4.28
npm ERR! code E404
npm ERR! not ok code 0

...

11:26:46 xtuple >> Creating systemd service unit using filename /etc/systemd/system/xtuple-xtuple-web.service

/etc/xtuple/4.10.1/xtuple-web/config.js:30
      port: ,
            ^
<<ERROR 2017-12-26T03:26:46.343Z>> Unrecoverable exception. Unexpected token ,SyntaxError: Unexpected token ,
    at Module._compile (module.js:439:25)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at exports.build (/opt/xtuple/4.10.1/xtuple-web/xtuple/scripts/lib/build_all.js:123:16)
    at /opt/xtuple/4.10.1/xtuple-web/xtuple/scripts/build_app.js:35:3
    at Object.<anonymous> (/opt/xtuple/4.10.1/xtuple-web/xtuple/scripts/build_app.js:55:2)
    at Module._compile (module.js:456:26)
11:26:46 xtuple >> All set! You should now be able to log on to this server at https://172.16.16.41: with username admin and password admin. Make sure you change your password!

--- a/common.sh
+++ b/common.sh
@@ -1,14 +1,14 @@
 #!/bin/bash
 
 do_exit() {
-echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
+echo "In: ${BASH_SOURCE} ${FUNCNAME[@]}"
 
     log "Exiting xTuple Admin Utility"
     exit 0
 }
 
 die() {
-echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
+  echo "In: ${BASH_SOURCE} ${FUNCNAME[@]}"
 
   TRAPMSG="$@"
   log $@
@@ -188,19 +188,14 @@ test_connection() {
 }
 
 setup_sudo() {
-echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
+  echo "In: ${BASH_SOURCE} ${FUNCNAME[@]}"
 
-if [[ ! -f "/etc/sudoers.d/${DEPLOYER_NAME}" ]] || ! grep -q "${DEPLOYER_NAME}" /etc/sudoers.d/${DEPLOYER_NAME}; then
- # echo "Setting up user: $DEPLOYER_NAME for sudo"
- # echo "You might be prompted for your password."
- # (echo "${DEPLOYER_NAME} ALL=(ALL) NOPASSWD:ALL" )| sudo tee -a /etc/sudoers.d/90-xtau-users >/dev/null
-
-  sudo printf "${DEPLOYER_NAME} ALL=(ALL) NOPASSWD: ALL\n" > /etc/sudoers.d/${DEPLOYER_NAME} && \
-  sudo chmod 440 /etc/sudoers.d/${DEPLOYER_NAME}
-else
-  echo "User: $DEPLOYER_NAME already setup in sudoers.d"
-fi
-
+  if [[ ! -f "/etc/sudoers.d/${DEPLOYER_NAME}" ]] || ! grep -q "${DEPLOYER_NAME}" /etc/sudoers.d/${DEPLOYER_NAME}; then
+    printf "${DEPLOYER_NAME} ALL=(ALL) NOPASSWD: ALL\n" | sudo tee -a /etc/sudoers.d/${DEPLOYER_NAME}
+    sudo chmod 440 /etc/sudoers.d/${DEPLOYER_NAME}
+  else
+    echo "User: $DEPLOYER_NAME already setup in sudoers.d"
+  fi
 }
 
 

--- a/common.sh
+++ b/common.sh
@@ -45,12 +45,12 @@ echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
 }
 
 # $1 is the URL
-# $3 is the output file name
+# $2 is the output file name
 dlf_fast_console() {
-echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
+    echo "In: ${BASH_SOURCE} ${FUNCNAME[0]} $@"
 
     log "Downloading $1 to file $2 using axel console output only"
-    axel -n 5 "$1" -o "$2" > /dev/null
+    axel --num-connections=5 --output="$2" "$1" > /dev/null
 }
 
 # $1 is the msg

--- a/config.sh
+++ b/config.sh
@@ -54,7 +54,7 @@ GITHUBPASS=
 # export CONFIG_DIR=$(pwd)/xdruple-server/config
 
 # export TYPE='server'
-# export DEPLOYER_NAME=`whoami`
+export DEPLOYER_NAME=$(whoami)
 # export TIMEZONE=America/New_York
 
 #sudo locale-gen en_US.UTF-8 && \

--- a/config.sh
+++ b/config.sh
@@ -28,7 +28,7 @@ NGINX_SITE=
 # auto populated if site exists, otherwise can be used to create a site
 NGINX_DOMAIN=
 NGINX_HOSTNAME=
-NGINX_PORT=
+NGINX_PORT=${NGINX_PORT:-8443}
 NGINX_CERT=
 NGINX_KEY=
 # generate new certs if the specified ones don't exist

--- a/database.sh
+++ b/database.sh
@@ -310,19 +310,24 @@ echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
     if [ $RET -ne 0 ]; then
         msgbox "Something has gone wrong. Check log and correct any issues."
         return $RET
-    else
-        log_exec psql -qAt -U $PGUSER -h $PGHOST -p $PGPORT -d $DATABASE -f "$XTDIR"/lib/orm/source/create_xt_schema.sql.sql
-        log_exec psql -qAt -U $PGUSER -h $PGHOST -p $PGPORT -d $DATABASE -f "$XTDIR"/lib/orm/source/create_plv8.sql
-        log "Restoring database $DATABASE from file $1 on server $PGHOST:$PGPORT"
-        pg_restore --username "$PGUSER" --port "$PGPORT" --host "$PGHOST" --dbname "$DATABASE" "$1" 2>restore_output.log
-        RET=$?
-        if [ $RET -ne 0 ]; then
-            msgbox "$(cat restore_output.log)"
-            return $RET
-        else
-            return 0
-        fi
     fi
+    # XTDIR/lib/orm/source/create_{xt_schema,plv8}.sql but XTDIR isn't installed yet
+    log_exec psql -qAt -U $PGUSER -h $PGHOST -p $PGPORT -d $DATABASE <<EOSCRIPT
+      CREATE SCHEMA IF NOT EXISTS xt;
+      GRANT ALL ON SCHEMA xt TO GROUP xtrole;
+      CREATE OR REPLACE FUNCTION xt.js_init(debug BOOLEAN DEFAULT false, initialize BOOLEAN DEFAULT false)
+      RETURNS VOID AS 'BEGIN RETURN; END;' LANGUAGE plpgsql;
+      CREATE EXTENSION IF NOT EXISTS plv8;
+EOSCRIPT
+    log "Restoring database $DATABASE from file $1 on server $PGHOST:$PGPORT"
+    pg_restore --username "$PGUSER" --port "$PGPORT" --host "$PGHOST" --dbname "$DATABASE" "$1" 2>restore_output.log
+    RET=$?
+    if [ $RET -ne 0 ]; then
+        msgbox "$(cat restore_output.log)"
+        return $RET
+    fi
+
+    return 0
 }
 
 # list the existing databases on the current configured cluster

--- a/database.sh
+++ b/database.sh
@@ -42,50 +42,50 @@ echo "In: ${BASH_SOURCE} ${FUNCNAME[0]} $*"
 # manual, prompt for location, don't delete
 # $1 where to save database to
 # $2 is version to grab
-# $3 is type of database to grab (empty, demo, manufacturing, distribution, masterref)
-# $4 is the database name to restore to
+# $3 is type of database: empty, demo, manufacturing, distribution, masterref, ...
 download_database() {
-echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
+    echo "In: ${BASH_SOURCE} ${FUNCNAME[0]} $@"
 
-    DBVERSION="$2"
+    local DESTFILE="$1"
+    local DBVERSION="$2"
+    local DBTYPE="${3:-$DBTYPE}"
+
     if [ -z "$DBVERSION" ]; then
         msgbox "Database version not specified."
         return 1
     fi
 
-    DBTYPE="${3:-$DBTYPE}"
     DBTYPE="${DBTYPE:-demo}"
 
-    DEMODEST="$1"
-    if [ -z "$DEMODEST" ] && [ "$MODE" = "manual" ]; then
-        DEMODEST=$(whiptail --backtitle "$( window_title )" --inputbox "Enter the filename where you would like to save the database" 8 60 3>&1 1>&2 2>&3)
+    if [ -z "$DESTFILE" ] && [ "$MODE" = "manual" ]; then
+        DESTFILE=$(whiptail --backtitle "$( window_title )" --inputbox "Enter the filename where you would like to save the database" 8 60 3>&1 1>&2 2>&3)
         RET=$?
         if [ $RET -ne 0 ]; then
             return $RET
         fi
-    elif [ -z "$DEMODEST" ]; then
+    elif [ -z "$DESTFILE" ]; then
         return 127
     fi
 
-    DB_URL="http://files.xtuple.org/$DBVERSION/$DBTYPE.backup"
-    MD5_URL="http://files.xtuple.org/$DBVERSION/$DBTYPE.backup.md5sum"
+    local DB_URL="http://files.xtuple.org/$DBVERSION/$DBTYPE.backup"
+    local MD5_URL="http://files.xtuple.org/$DBVERSION/$DBTYPE.backup.md5sum"
 
-    log "Saving "$DB_URL" as "$DEMODEST"."
+    log "Saving $DB_URL as $DESTFILE."
+    mkdir -p $(dirname $DESTFILE)
     if [ $MODE = "auto" ]; then
-        dlf_fast_console $DB_URL "$DEMODEST"
-        dlf_fast_console $MD5_URL "$DEMODEST".md5sum
+        dlf_fast_console $DB_URL "$DESTFILE"
+        dlf_fast_console $MD5_URL "$DESTFILE".md5sum
     else
-        dlf_fast $DB_URL "Downloading Demo Database. Please Wait." "$DEMODEST"
-        dlf_fast $MD5_URL "Downloading MD5SUM. Please Wait." "$DEMODEST".md5sum
+        dlf_fast $DB_URL  "Downloading $DBTYPE Database. Please Wait." "$DESTFILE"
+        dlf_fast $MD5_URL "Downloading MD5SUM. Please Wait." "$DESTFILE".md5sum
     fi
 
-    VALID=`cat "$DEMODEST".md5sum | awk '{printf $1}'`
-    CURRENT=`md5sum "$DEMODEST" | awk '{printf $1}'`
+    local VALID=$(cat "$DESTFILE".md5sum | awk '{printf $1}')
+    local CURRENT=$(md5sum "$DESTFILE" | awk '{printf $1}')
     if [ "$VALID" != "$CURRENT" ]; then
         msgbox "There was an error verifying the downloaded database."
         return 1
     fi
-
 }
 
 # Copies database $1 to $2 by backing up $1 and restoring to $2
@@ -208,7 +208,7 @@ echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
 # or restore a local file
 # This can not be run in automatic mode
 create_database() {
-echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
+    echo "In: ${BASH_SOURCE} ${FUNCNAME[0]} $@"
 
     [ "$MODE" = "auto" ] && return 127
 
@@ -222,7 +222,7 @@ echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
     done < <( ls -t "${DATABASEDIR}/*.backup" | tr ' ' '_' )
     BACKUPDBS=()
     while read -r line ; do
-	    BACKUPDBS+=("$line" "$line")
+        BACKUPDBS+=("$line" "$line")
     done < <( ls -t $BACKUPDIR | awk '{printf("Restore %s\n", $0)}' | tr ' ' '_' )
     CUSTOMDB=("EnterFileName..." "EnterFileName...")
 
@@ -237,9 +237,8 @@ echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
         return $RET
     fi
 
-    echo $CHOICE | grep '^Download'
-	if [ $? -eq 0 ]; then
-	    DBVERSION=$(echo $CHOICE | grep -oP '\d\.\d\d?\.\d')
+    if echo $CHOICE | grep '^Download' ; then
+        DBVERSION=$(echo $CHOICE | grep -oP '\d\.\d\d?\.\d')
         EDITIONS=()
         while read line ; do
             EDITIONS+=("$line" "$line")
@@ -253,19 +252,16 @@ echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
         fi
 
         EDITION=$(echo $CHOICE | cut -f1 -d'.')
-	    download_database "$DATABASEDIR/$EDITION_$DBVERSION.backup" "$DBVERSION" "$EDITION"
-        restore_database "$DATABASEDIR/$EDITION_$DBVERSION.backup"
-	    return $?
-	fi
-
-	echo $CHOICE | grep '^Backup db'
-	if [ $? -eq 0 ]; then
-	    DATABASE=$(echo $CHOICE | sed 's/Backup db //')
-	    restore_database "$BACKUPDIR/$DATABASE"
+        download_database "$DATABASEDIR/$DBVERSION/$EDITION.backup" "$DBVERSION" "$EDITION"
+        restore_database  "$DATABASEDIR/$DBVERSION/$EDITION.backup"
         return $?
-	fi
 
-    if [ "$CHOICE" = "EnterFileName..." ]; then
+    elif echo $CHOICE | grep '^Backup db' ; then
+        DATABASE=$(echo $CHOICE | sed 's/Backup db //')
+        restore_database "$BACKUPDIR/$DATABASE"
+        return $?
+
+    elif [ "$CHOICE" = "EnterFileName..." ]; then
         DATABASE=$(whiptail --backtitle "$( window_title )" --inputbox "Full Database Pathname" 8 60 `pwd` 3>&1 1>&2 2>&3)
         RET=$?
         if [ $RET -ne 0 ]; then
@@ -276,8 +272,8 @@ echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
         return $?
     fi
 
-	restore_database "$DATABASEDIR/$CHOICE"
-	return $?
+    restore_database "$DATABASEDIR/$CHOICE"
+    return $?
 
 }
 

--- a/database.sh
+++ b/database.sh
@@ -311,6 +311,8 @@ echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
         msgbox "Something has gone wrong. Check log and correct any issues."
         return $RET
     else
+        log_exec psql -qAt -U $PGUSER -h $PGHOST -p $PGPORT -d $DATABASE -f "$XTDIR"/lib/orm/source/create_xt_schema.sql.sql
+        log_exec psql -qAt -U $PGUSER -h $PGHOST -p $PGPORT -d $DATABASE -f "$XTDIR"/lib/orm/source/create_plv8.sql
         log "Restoring database $DATABASE from file $1 on server $PGHOST:$PGPORT"
         pg_restore --username "$PGUSER" --port "$PGPORT" --host "$PGHOST" --dbname "$DATABASE" "$1" 2>restore_output.log
         RET=$?

--- a/mobileclient.sh
+++ b/mobileclient.sh
@@ -1,5 +1,39 @@
 #!/bin/bash
 
+gitco() {
+  local DESTDIR="$1"
+  local REPO="$(basename $2 .git)"
+  local REFSPEC="$3"
+  local GITHUBNAME="${4:-$GITHUBNAME}"
+  local GITHUBPASS="${5:-$GITHUBPASS}"
+
+  local STARTDIR=$(pwd)
+  local GHUSERSPEC=
+
+  if [ -n "$GITHUBPASS" ] ; then
+    GHUSERSPEC="$GITHUBNAME:$GITHUBPASS@"
+  elif [ -n "$GITHUBNAME" ] ; then
+    GHUSERSPEC="$GITHUBNAME@"
+  fi
+
+  mkdir -p "$DESTDIR"
+  cd "$DESTDIR"
+  ls $REPO || echo "$REPO not found"
+  if [ ! -d $REPO/.git ] ; then
+    git clone https://${GHUSERSPEC}github.com/xtuple/$REPO.git || die
+  fi
+  cd $REPO
+  if ! git remote -v | grep -q xtuple/$REPO ; then
+    git remote add XTUPLE https://github.com/xtuple/$REPO.git
+    git fetch https://${GHUSERSPEC}github.com/xtuple/$REPO.git
+  fi
+
+  git checkout $REFSPEC                           || die
+  git submodule update --init --recursive         || die
+  npm install                                     || die
+  cd "$STARTDIR"
+}
+
 mwc_menu() {
 echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
 
@@ -25,7 +59,6 @@ echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
             esac
         fi
     done
-
 }
 
 install_mwc_menu() {
@@ -173,11 +206,7 @@ install_mwc() {
     log "Installing node 0.10.40..."
     log_exec sudo n 0.10.40                                                     || die
 
-    # need to install npm of course...why doesn't 2.3.14 exist?
-    log_exec sudo npm install -g npm@1.4.28                                     || die
-
-    # cleanup existing folder or git will throw a hissy fit
-    sudo rm -rf /opt/xtuple/$MWCVERSION/"$MWCNAME"
+    log_exec sudo npm install -g npm@2.x.x                                      || die
 
     log "Cloning xTuple Web Client Source Code to /opt/xtuple/$MWCVERSION/xtuple"
     log "Using version $MWCVERSION with the given name $MWCNAME"
@@ -185,23 +214,13 @@ install_mwc() {
     log_exec sudo chown -R ${DEPLOYER_NAME}.${DEPLOYER_NAME} /opt/xtuple        || die
 
     # main code
-    git clone https://github.com/xtuple/xtuple.git \
-                                   /opt/xtuple/$MWCVERSION/$MWCNAME/xtuple      || die
-    cd /opt/xtuple/$MWCVERSION/$MWCNAME/xtuple                                  || die
-    git checkout $MWCREFSPEC                                                    || die
-    git submodule update --init --recursive                                     || die
+    gitco /opt/xtuple/$MWCVERSION/$MWCNAME xtuple $MWCREFSPEC
+    cd /opt/xtuple/$MWCVERSION/$MWCNAME/xtuple
     npm install bower                                                           || die
-    npm install                                                                 || die
-
     # private extensions
     if $PRIVATEEXT ; then
         log "Installing the commercial extensions"
-        git clone https://$GITHUBNAME:$GITHUBPASS@github.com/xtuple/private-extensions.git \
-                            /opt/xtuple/$MWCVERSION/$MWCNAME/private-extensions || die
-        cd /opt/xtuple/$MWCVERSION/$MWCNAME/private-extensions                  || die
-        git checkout $MWCREFSPEC                                                || die
-        git submodule update --init --recursive                                 || die
-        npm install                                                             || die
+        gitco /opt/xtuple/$MWCVERSION/$MWCNAME private-extensions $MWCREFSPEC $GITHUBNAME $GITHUBPASS
     else
         log "Not installing the commercial extensions"
     fi
@@ -292,8 +311,6 @@ install_mwc() {
     fi
 
     log_exec cd $XTDIR                                                               || die
-    log_exec psql -qAt -U $PGUSER -h $PGHOST -p $PGPORT -d $ERP_DATABASE_NAME -f lib/orm/source/create_xt_schema.sql
-    log_exec psql -qAt -U $PGUSER -h $PGHOST -p $PGPORT -d $ERP_DATABASE_NAME -f lib/orm/source/create_plv8.sql
     log_exec ./scripts/build_app.js -c /etc/xtuple/$MWCVERSION/"$MWCNAME"/config.js  || die
 
     # now that we have the script, start the server!

--- a/mobileclient.sh
+++ b/mobileclient.sh
@@ -39,7 +39,7 @@ echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
         fi
 
         get_database_list
-    
+
         if [ -z "$DATABASES" ]; then
             msgbox "No databases detected on this system"
             return 1
@@ -139,9 +139,9 @@ echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
 # $6 is github username
 # $7 is github password
 install_mwc() {
-echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
+    echo "In: ${BASH_SOURCE} ${FUNCNAME[0]} $@"
 
-    log "Installing web client"
+    log "Web enabling"
 
     MWCVERSION="${1:-$MWCVERSION}"
     MWCREFSPEC="${2:-$MWCREFSPEC}"
@@ -151,14 +151,12 @@ echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
     PRIVATEEXT="${PRIVATEEXT:-false}"
     if [ ! "$PRIVATEEXT" = "true" ]; then
         PRIVATEEXT=false
-    else
-        PRIVATEEXT=true
     fi
 
     ERP_DATABASE_NAME="${5:-$ERP_DATABASE_NAME}"
     if [ -z "$ERP_DATABASE_NAME" ]; then
         log "No database name passed to install_mwc... exiting."
-        do_exit
+        die
     fi
 
     GITHUBNAME="${6:-$GITHUBNAME}"
@@ -166,43 +164,51 @@ echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
 
     setup_webprint
 
-    log "Creating xtuple user..."
-    log_exec sudo useradd xtuple -m -s /bin/bash
-
     log "Installing n..."
-    cd $WORKDIR    
+    cd $WORKDIR
     wget https://raw.githubusercontent.com/visionmedia/n/master/bin/n -qO n
-    log_exec chmod +x n
-    log_exec sudo mv n /usr/bin/n
+    log_exec chmod +x n                                                         || die
+    log_exec sudo mv n /usr/bin/n                                               || die
     # use it to set node to 0.10
     log "Installing node 0.10.40..."
-    log_exec sudo n 0.10.40
+    log_exec sudo n 0.10.40                                                     || die
 
     # need to install npm of course...why doesn't 2.3.14 exist?
-    log_exec sudo npm install -g npm@1.4.28
+    log_exec sudo npm install -g npm@1.4.28                                     || die
 
     # cleanup existing folder or git will throw a hissy fit
     sudo rm -rf /opt/xtuple/$MWCVERSION/"$MWCNAME"
 
     log "Cloning xTuple Web Client Source Code to /opt/xtuple/$MWCVERSION/xtuple"
     log "Using version $MWCVERSION with the given name $MWCNAME"
-    log_exec sudo mkdir -p /opt/xtuple/$MWCVERSION/"$MWCNAME"
-    log_exec sudo chown -R ${DEPLOYER_NAME}.${DEPLOYER_NAME} /opt/xtuple
+    log_exec sudo mkdir -p /opt/xtuple/$MWCVERSION/"$MWCNAME"                   || die
+    log_exec sudo chown -R ${DEPLOYER_NAME}.${DEPLOYER_NAME} /opt/xtuple        || die
 
     # main code
-    log_exec sudo su - xtuple -c "cd /opt/xtuple/$MWCVERSION/"$MWCNAME" && git clone https://github.com/xtuple/xtuple.git && cd  /opt/xtuple/$MWCVERSION/"$MWCNAME"/xtuple && git checkout $MWCREFSPEC && git submodule update --init --recursive && npm install bower && npm install"
-#    log_exec sudo su - xtuple -c "cd /opt/xtuple/$MWCVERSION/"$MWCNAME" && git clone https://github.com/xtuple/xtuple.git && cd  /opt/xtuple/$MWCVERSION/"$MWCNAME"/xtuple && git checkout $MWCREFSPEC && git submodule update --init --recursive"
+    git clone https://github.com/xtuple/xtuple.git \
+                                   /opt/xtuple/$MWCVERSION/$MWCNAME/xtuple      || die
+    cd /opt/xtuple/$MWCVERSION/$MWCNAME/xtuple                                  || die
+    git checkout $MWCREFSPEC                                                    || die
+    git submodule update --init --recursive                                     || die
+    npm install bower                                                           || die
+    npm install                                                                 || die
+
     # private extensions
-    if [ $PRIVATEEXT = "true" ]; then
+    if $PRIVATEEXT ; then
         log "Installing the commercial extensions"
-        log_exec sudo su xtuple -c "cd /opt/xtuple/$MWCVERSION/"$MWCNAME" && git clone https://"$GITHUBNAME":"$GITHUBPASS"@github.com/xtuple/private-extensions.git && cd /opt/xtuple/$MWCVERSION/"$MWCNAME"/private-extensions && git checkout $MWCREFSPEC && git submodule update --init --recursive && npm install"
+        git clone https://$GITHUBNAME:$GITHUBPASS@github.com/xtuple/private-extensions.git \
+                            /opt/xtuple/$MWCVERSION/$MWCNAME/private-extensions || die
+        cd /opt/xtuple/$MWCVERSION/$MWCNAME/private-extensions                  || die
+        git checkout $MWCREFSPEC                                                || die
+        git submodule update --init --recursive                                 || die
+        npm install                                                             || die
     else
         log "Not installing the commercial extensions"
     fi
 
     if [ ! -f /opt/xtuple/$MWCVERSION/"$MWCNAME"/xtuple/node-datasource/sample_config.js ]; then
         log "Hrm, sample_config.js doesn't exist.. something went wrong. check the output/log and try again"
-        do_exit
+        die
     fi
 
     export XTDIR=/opt/xtuple/$MWCVERSION/"$MWCNAME"/xtuple
@@ -282,15 +288,11 @@ echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
         esac
     else
         log "Seriously? We made it all the way to where I need to write out the init script and suddenly I can't detect your distro -> $DISTRO codename -> $CODENAME"
-        do_exit
+        die
     fi
 
-    log_exec sudo su - xtuple -c "cd $XTDIR && ./scripts/build_app.js -c /etc/xtuple/$MWCVERSION/"$MWCNAME"/config.js"
-    RET=$?
-    if [ $RET -ne 0 ]; then
-        log "buildapp failed to run. Check output and try again"
-        do_exit
-    fi
+    log_exec cd $XTDIR                                                               || die
+    log_exec ./scripts/build_app.js -c /etc/xtuple/$MWCVERSION/"$MWCNAME"/config.js  || die
 
     # now that we have the script, start the server!
     if [ $DISTRO = "ubuntu" ]; then
@@ -317,7 +319,7 @@ echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
         esac
     else
         log "Seriously? We made it all the way to where I need to start the server and suddenly I can't detect your distro -> $DISTRO codename -> $CODENAME"
-        do_exit
+        die
     fi
 
     # assuming etho for now... hostname -I will give any non-local address if we wanted

--- a/mobileclient.sh
+++ b/mobileclient.sh
@@ -292,6 +292,8 @@ install_mwc() {
     fi
 
     log_exec cd $XTDIR                                                               || die
+    log_exec psql -qAt -U $PGUSER -h $PGHOST -p $PGPORT -d $ERP_DATABASE_NAME -f lib/orm/source/create_xt_schema.sql
+    log_exec psql -qAt -U $PGUSER -h $PGHOST -p $PGPORT -d $ERP_DATABASE_NAME -f lib/orm/source/create_plv8.sql
     log_exec ./scripts/build_app.js -c /etc/xtuple/$MWCVERSION/"$MWCNAME"/config.js  || die
 
     # now that we have the script, start the server!

--- a/nginx.sh
+++ b/nginx.sh
@@ -110,7 +110,9 @@ type nginx >/dev/null 2>&1 || { echo >&2 "nginx not installed, installing"; inst
 #   NGINX_KEY       - website key with the same requirements
 #   NGINX_PORT      - nginx port to listen to
 configure_nginx() {
-echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
+    echo "In: ${BASH_SOURCE} ${FUNCNAME[0]} $@"
+
+    local CURRENTDIR=$(pwd)
 
     log "Configuring nginx"
 
@@ -118,9 +120,11 @@ echo "In: ${BASH_SOURCE} ${FUNCNAME[0]}"
     [[ -e /etc/nginx/sites-available/default ]] && sudo rm /etc/nginx/sites-available/default /etc/nginx/sites-enabled/default 2>&1 >/dev/null
 
     log "Creating site file"
+    cd "$WORKING"
     sudo cp templates/nginx-site /etc/nginx/sites-available/$NGINX_SITE
     sudo sed -i -e "s#DOMAINNAME#$NGINX_DOMAIN#" -e "s#HOSTNAME#$NGINX_HOSTNAME#" /etc/nginx/sites-available/$NGINX_SITE
     RET=$?
+    cd "$CURRENTDIR"
     if [ $RET -ne 0 ]; then
         msgbox "Error configuring nginx.  Check site file in /etc/nginx/sites-available"
         return $RET

